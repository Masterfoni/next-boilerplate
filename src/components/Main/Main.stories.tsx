import { Story, Meta } from '@storybook/react/types-6-0';
import { ReactElement } from 'react';
import Main from '.';

export default {
  title: 'Main',
  component: Main
} as Meta;

export const Basic: Story = (args): ReactElement => <Main {...args} />;
