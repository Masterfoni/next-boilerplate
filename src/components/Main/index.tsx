import { ReactElement } from 'react';

const Main = (): ReactElement => (
  <main>
    <h1>{'boilerplate first test'}</h1>
  </main>
);

export default Main;
