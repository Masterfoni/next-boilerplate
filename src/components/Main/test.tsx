import { render, screen } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render', () => {
    const { container } = render(<Main />);

    expect(container).toMatchSnapshot();
  });

  it('should render heading', () => {
    render(<Main />);

    expect(
      screen.getByRole('heading', { name: /boilerplate first test/i })
    ).toBeInTheDocument();
  });
});
