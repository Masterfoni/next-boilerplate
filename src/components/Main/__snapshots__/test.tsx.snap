// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render 1`] = `
<div>
  <main>
    <h1>
      boilerplate first test
    </h1>
  </main>
</div>
`;
